src/System.Security.Cryptography.Primitives/tests/ZeroMemoryTests.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Security.Cryptography.Primitives.Tests
+{
+    public static class ZeroMemoryTests
+    {
+        [Theory]
+        [InlineData(1)]
+        [InlineData(128 / 8)]
+        [InlineData(256 / 8)]
+        [InlineData(512 / 8)]
+        [InlineData(96)]
+        [InlineData(1024)]
+        public static void MemoryGetsCleared(int byteLength)
+        {
+            byte[] rented = ArrayPool<byte>.Shared.Rent(byteLength);
+            Span<byte> testSpan = new Span<byte>(rented, 0, byteLength);
+
+            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+            {
+                bool hasData = false;
+
+                // i should really only iterate when byteLength is 1, and then
+                // only 1/256 executions.
+                //
+                // The chances of this failing are 1 in 1.2e24, unless the RNG is broken.
+                for (int i = 0; i < 10 && !hasData; i++)
+                {
+                    rng.GetBytes(testSpan);
```

Comment:
You don't want to just use Fill? :smile:

Reply:
Drat, missed one :).  Fill was written last :).